packages/x-data-grid/src/colDef/gridSingleSelectOperators.ts

```diff
@@ -10,6 +11,20 @@ const parseObjectValue = (value: unknown) => {
   return value.value;
 };
 
+const createAnyFilterFn = (negate: boolean) => (filterItem: GridFilterItem) => {
+  if (!Array.isArray(filterItem.value) || filterItem.value.length === 0) {
+    return null;
+  }
+  const filterItemValues = filterItem.value.map(parseObjectValue);
+  return (value: unknown): boolean => {
+    if (value == null) {
+      return negate;
+    }
+    const isAnyOf = filterItemValues.includes(parseObjectValue(value));
+    return negate ? !isAnyOf : isAnyOf;
```

Comment:
Move the negation condition outside the returned callback, it could make a difference with large datasets.